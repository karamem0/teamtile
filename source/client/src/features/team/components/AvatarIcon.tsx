//
// Copyright (c) 2021-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { AvatarSize } from '@fluentui/react-components';

import Presenter from './AvatarIcon.presenter';

interface AvatarIconProps {
  icon?: string,
  name?: string,
  size?: AvatarSize
}

function AvatarIcon(props: Readonly<AvatarIconProps>) {

  const {
    icon,
    name,
    size
  } = props;

  return (
    <Presenter
      icon={icon}
      name={name}
      size={size} />
  );

}

export default AvatarIcon;
