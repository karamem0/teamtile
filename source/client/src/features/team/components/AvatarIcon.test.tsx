//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { render, screen } from '@testing-library/react';

import AvatarIcon from './AvatarIcon.presenter';

test('create shapshot of when url is undefined', async () => {
  const params = {
    icon: undefined,
    name: 'HR Taskforce',
    size: undefined
  };
  render(<AvatarIcon {...params} />);
  expect((await screen.findAllByText(/^.*$/))[0]).toMatchSnapshot();
});

test('create shapshot of when url is not undefined', async () => {
  const params = {
    icon: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAAEnQAABJ0Ad5mH3gAAAAMSURBVBhXY/j//z8ABf4C/qc1gYQAAAAASUVORK5CYII=',
    name: 'HR Taskforce',
    size: undefined
  };
  render(<AvatarIcon {...params} />);
  expect((await screen.findAllByText(/^.*$/))[0]).toMatchSnapshot();
});
