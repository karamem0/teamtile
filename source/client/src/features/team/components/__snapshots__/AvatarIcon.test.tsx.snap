// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create shapshot of when url is not undefined 1`] = `
<body>
  <div>
    <span
      aria-label="HR Taskforce"
      class="fui-Avatar rz243wu ___1osyt9h_13rrzcx f11d4kpn f18f03hv f1uuiafn"
      id="avatar-r1"
      role="img"
    >
      <span
        class="fui-Avatar__initials rip04v"
        id="avatar-r1__initials"
      >
        HT
      </span>
      <img
        alt=""
        aria-hidden="true"
        class="fui-Avatar__image r136dc0n"
        role="presentation"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAAEnQAABJ0Ad5mH3gAAAAMSURBVBhXY/j//z8ABf4C/qc1gYQAAAAASUVORK5CYII="
      />
    </span>
  </div>
</body>
`;

exports[`create shapshot of when url is undefined 1`] = `
<body>
  <div>
    <span
      aria-label="HR Taskforce"
      class="fui-Avatar rz243wu ___1osyt9h_13rrzcx f11d4kpn f18f03hv f1uuiafn"
      id="avatar-r0"
      role="img"
    >
      <span
        class="fui-Avatar__initials rip04v"
        id="avatar-r0__initials"
      >
        HT
      </span>
      <img
        alt=""
        aria-hidden="true"
        class="fui-Avatar__image r136dc0n"
        role="presentation"
      />
    </span>
  </div>
</body>
`;
